go/hamming/hamming.go

```diff
@@ -1,4 +1,37 @@
+// This package implements an hamming distance calculation function
+// aimed at DNA strands
+
 package hamming
 
-func Distance(a, b string) (int, error) {
+import (
+	"errors"
+	"strings"
+)
+
+/*  Distance returns the hamming distance between two dna strands
+    passed as string parameters. Returns -1 if the strands are of
+	different lengths.
+    parameters :
+		a : strand as a string
+		b : strand as a string
+*/
+func Distance(a, b string) (distance int, distanceError error) {
+	if len(a) != len(b) {
+		distance = -1
```

Comment:
do you need the `-1` value (I think the tests should be ok without it) given that you are returning an error? Generally if there is an error, all the other data is considered garbage and should not be relied upon. Therefore, the default value of 0 might be good enough...